File: LV0/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
  let x = 0;
  while (true) {
    const transformedArr = arr.map((num) =>
      num >= 50 && num % 2 === 0
        ? num / 2
        : num < 50 && num % 2 === 1
        ? num * 2 + 1
        : num
    );

    if (arr.every((v, i) => v === transformedArr[i])) {
      return x;
    }

    arr = transformedArr;
    x++;
  }
}

File: LV0/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    x = 0
    while True:
        transformedArr = list(map(lambda num: num // 2 if num >= 50 and num % 2 == 0 else num * 2 + 1 if num < 50 and num % 2 == 1 else num, arr))
        if arr == transformedArr:
            return x
        arr = transformedArr
        x += 1
